### PR TITLE
Repair 'Announcements' plug-in

### PR DESCRIPTION
**Pull request by GriefMoDz**
### Plugin Changelog (v0.1.1) :: Announcements
index.js:
* Add notice types to current announcements (first-welcome & pewdiepie).
* Modify message content, button text and `onClick()` method for `pc-pewdiepie`.
* Announcements that are set to `alwaysDisplay` no longer write to the "dismissed" array found in settings.
* Update query selectors to use `.pc-wrapper` class instead of `.pc-guildWrapper` as it is now obsolete.

Notice.jsx:
* Add an additional notice type based around `noticeInfo` that changes the CTA button text to black on hover.

### CODE REVIEW

File: src/Powercord/plugins/pc-announcements/index.js
Comment (cyyynthia) [inline src/Powercord/plugins/pc-announcements/index.js]: why removing this? It's used in _renderNotice

File: src/Powercord/plugins/pc-announcements/index.js
Comment (cyyynthia) [inline src/Powercord/plugins/pc-announcements/index.js]: Then you can remove the alwaysDisplay parameter and let developers put the parameter in the notice itself

File: src/Powercord/plugins/pc-announcements/index.js
Comment (cyyynthia) [inline src/Powercord/plugins/pc-announcements/index.js]: this will prevent *any* notice to be added to the dismissed store if any announcement have alwaysDisplay on, even if the notice don't have it. You need to grab the notice using its ID (before we remove it from the store), and then check for the value.

File: src/Powercord/plugins/pc-announcements/index.js
Comment (cyyynthia) [inline src/Powercord/plugins/pc-announcements/index.js]: why put the require here?

**Review by cyyynthia — CHANGES_REQUESTED** [review]

File: src/Powercord/plugins/pc-announcements/index.js
Comment (cyyynthia) [inline src/Powercord/plugins/pc-announcements/index.js]: eslint won't like this `,` 👀
```suggestion
        alwaysDisplay: true
```

**Review by cyyynthia — CHANGES_REQUESTED** [review]

File: src/Powercord/plugins/pc-announcements/index.js
Comment (GriefMoDz) [reply src/Powercord/plugins/pc-announcements/index.js]: > why removing this? It's used in _renderNotice

No need to be alarmed. I'm aware that this constant variable is used in `_renderNotice` thus I've only went ahead and ~~re~~moved it to the location of said method (reasonings have been stated below).

File: src/Powercord/plugins/pc-announcements/index.js
Comment (GriefMoDz) [reply src/Powercord/plugins/pc-announcements/index.js]: > why put the require here?

To answer **why** I've chosen to move the require here compared to on `Line 10` is mainly due to the fact that the `Notice.TYPES` store is called too early (way before the notice classes from `getModule([ 'noticeBrand' ])` are even initialized).

That said, with the constant variable staying in it's original spot (on line 10) you'll find that the classes for announcements/notices can't be retrieved, causing them to be `undefined` and ultimately preventing styling (CSS) to both the `<div>` and its corresponding elements (`<button>` and `<Clickable>`).

Comment (cyyynthia) [reply src/Powercord/plugins/pc-announcements/index.js]: I see why the require was moved, but since _renderNotice can still be called early it can still cause race conditions in case of slow boots. In cases where the module is not initialized, Powercord will return a promise and it breaks as it's not what you'd expect.

A workarround can be to not pass classes directly in Notice.TYPES but just give constants (but keep the object for autocomplete/compatibility purposes) (`BLURPLE: 'BLURPLE'`) and mount classes in the `componentDidMount` with await and all promise stuff. Just use setState to store classes and make the component re-render, and boom.

What you can also do is save it in a variable and initialize store with it immediately, so if the module already have been fetched, it won't fetch it again.

Comment (GriefMoDz) [reply src/Powercord/plugins/pc-announcements/index.js]: Marking this conversation as **Resolved** - workaround can be found in the latest commit. Let me know if anything else needs to be changed.

File: src/Powercord/plugins/pc-announcements/Notice.jsx
Comment (aetheryx) [inline src/Powercord/plugins/pc-announcements/Notice.jsx]: This Promise.all() call seems unnecessary to me. `classes` is already an awaited variable, i.e. a resolved promise. You should be able to kill `Promise.all([ classes ]).then(value => { ... })` and replace all instances of `value[0]` with `classes`.

**Review by aetheryx — CHANGES_REQUESTED** [review]

File: src/Powercord/plugins/pc-announcements/Notice.jsx
Comment (GriefMoDz) [reply src/Powercord/plugins/pc-announcements/Notice.jsx]: That's a good call! Honestly not sure why I chose to use the `Promise.all()` call when there's only ever one key returned from the `noticeBrand` module. Anyway, I've gone ahead and added these requested changes to my latest commit.

**Review by cyyynthia — APPROVED** [review]
I pushed some changes to the PR (as discussed in Discord), now waiting for aeth's approval 👀

**Review by aetheryx — APPROVED** [review]

**Pull request merged**
